devdoc/watchdog.md

```diff
@@ -71,11 +82,15 @@ MOCKABLE_FUNCTION(, void, watchdog_reset, WATCHDOG_HANDLE, watchdog);
 
 **SRS_WATCHDOG_42_031: [** If `watchdog` is `NULL` then `watchdog_reset` shall return. **]**
 
-**SRS_WATCHDOG_42_032: [** `watchdog_reset` shall set the state of the watchdog to `STOP`. **]**
+**SRS_WATCHDOG_45_011: [** `watchdog_reset` shall call `sm_close_begin`. **]**
 
 **SRS_WATCHDOG_42_033: [** `watchdog_reset` shall cancel the current timer by calling `threadpool_timer_cancel`. **]**
 
-**SRS_WATCHDOG_42_034: [** `watchdog_reset` shall set the state of the watchdog to `RUNNING`. **]**
+**SRS_WATCHDOG_45_012: [** `watchdog_reset` shall call `sm_close_end`. **]**
+
+**SRS_WATCHDOG_45_013: [** `watchdog_reset` shall call `sm_open_begin`. **]**

```

Comment:
>l `sm_open_beg [](http://example.com/codeflow?start=53&length=14)

What happens if sm_close_begin fails? Or sm_open_begin? We should spec that and test it.

Reply:
Offline conversations with Matt suggest that if `close_begin_fails`, it's because it's already closed, and that's OK. It does implies that 2 calls to `watchdog_reset` might race - maybe I'll add a test for that.